lib/storage-service.js

```diff
@@ -53,6 +53,29 @@ function StorageService(options) {
   if (options.maxFieldsSize) {
     this.maxFieldsSize = options.maxFieldsSize;
   }
+  // pass through AWS specific options
+  // See http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property
+  if (options.StorageClass) {
+    this.StorageClass = options.StorageClass;
+  }
+  if (options.CacheControl) {
+    this.CacheControl = options.CacheControl;
+  }
+  if (options.ServerSideEncryption) {
+    this.ServerSideEncryption = options.ServerSideEncryption;
+  }
+  if (options.SSEKMSKeyId) {
+    this.SSEKMSKeyId = options.SSEKMSKeyId;
+  }
+  if (options.SSECustomerAlgorithm) {
+    this.SSECustomerAlgorithm = options.SSECustomerAlgorithm;
+  }
+  if (options.SSECustomerKey) {
+    this.SSECustomerKey = options.SSECustomerKey;
+  }
+  if (options.SSECustomerKeyMD5) {
+    this.SSECustomerKeyMD5 = options.SSECustomerKeyMD5;
+  }
```

Comment:
Is it better to organize such options as `options.aws.*`?

Reply:
It might be, but this is just a bug fix, not adding in a new feature. Those options are already there but not pulled through from the global datasources config at the moment.

Reply:
Maybe at least do the following?

```js
const awsOptionNames = ['SSECustomerKeyMD5', ...];
for(const awsOption of awsOptionNames) {
 /// ...
}
```

Reply:
I agree there's more elegant ways of doing it. I just used the same syntax as used in https://github.com/strongloop/loopback-component-storage/blob/e76d571a8c90cc4a1c50f2bf3966edaaf79eec40/lib/storage-handler.js#L150 which uses them downstream.

I think if we change one we'd have to change both.

Reply:
It would be nice to change both places.

Reply:
I've added some further commits to implement this and fix a couple of typos I noticed.